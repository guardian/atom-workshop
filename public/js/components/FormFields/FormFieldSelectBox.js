import React, {PropTypes} from 'react';
import ShowErrors from '../Utilities/ShowErrors';
import { errorPropType } from '../../constants/errorPropType';

export default class FormFieldSelectBox extends React.Component {

  static propTypes = {
    fieldLabel: PropTypes.string,
    fieldName: PropTypes.string,
    fieldValue: PropTypes.string,
    fieldErrors: PropTypes.arrayOf(errorPropType),
    selectValues: PropTypes.array,
    onUpdateField: PropTypes.func
  };

  renderOption(option) {
    return (
        <option key={option} value={option}>{option}</option>
    );
  }

  onUpdate = (e) => {
    this.props.onUpdateField(e.target.value);
  }


  renderOptions() {
    return this.props.selectValues.map(this.renderOption);
  }

  render() {
    return (
        <div>
          <label htmlFor={this.props.fieldName} className="form__label">{this.props.fieldLabel}</label>
          <select className="form__field form__field--select" value={this.props.fieldValue} onChange={this.onUpdate}>
            {this.renderOptions()}
          </select>
          {this.props.fieldErrors && this.props.fieldErrors.length ? <ShowErrors errors={this.props.fieldErrors}/>  : false}
        </div>
    );
  }
}
