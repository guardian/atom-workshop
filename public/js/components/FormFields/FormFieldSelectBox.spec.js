import React from 'react';
import FormFieldSelectBox from './FormFieldSelectBox';
import renderer from 'react-test-renderer';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

let fieldLabel = 'test',
    fieldName = 'test',
    fieldValue= 'test',
    selectValues = ['test', 'test1'];

test('Should render', () => {

  const updateFn = jest.fn();
  const component = renderer.create(
    <FormFieldSelectBox fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} selectValues={selectValues} onUpdateField={updateFn} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Should call update function on change', () => {

  const updateFn = jest.fn();
  const select = shallow(
    <FormFieldSelectBox fieldLabel={fieldLabel} fieldName={fieldName} fieldValue={fieldValue} selectValues={selectValues} onUpdateField={updateFn} />
  );

  select.find('select').simulate('change', {target: {value: "test"}});

  expect(updateFn).toHaveBeenCalledTimes(1);
});
