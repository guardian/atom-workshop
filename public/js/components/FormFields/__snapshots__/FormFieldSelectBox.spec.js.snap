exports[`test Should render 1`] = `
<div>
  <label
    className="form__label"
    htmlFor="test">
    test
  </label>
  <select
    className="form__field form__field--select"
    onChange={[Function]}
    value="test">
    <option
      value="test">
      test
    </option>
    <option
      value="test1">
      test1
    </option>
  </select>
  <div />
</div>
`;
